Apply ARM fix to devtools-archlinuxcn

yuyichao opened the pull request:
The new patch added (`0004-*`) fixes the handling of server URL on arm without breaking it on x86 (AFAICT....) The patches applied are tracked at https://github.com/yuyichao/devtools/tree/archlinuxcn right now.

With this, on ARM, one should just install devtools-archlinuxcn to get a working base devtools. devtools-arm-git should work for providing a working arm wrapper (`extra-*-build`) on top of it, and `devtools-cn-git` should provide all the `archlinuxcn-*` wrappers.

yan12125 commented:
Thank you very much for this. I'm also working on updating devtools-archlinuxcn. Some discussions are needed, especially for the new BUILDTOOL variable. Could you wait me for a few more days before merging this? I need to collect related information.

Some other related discussions: https://github.com/archlinuxcn/repo/pull/2789#issuecomment-1140229167

CoelacanthusHex commented on archlinuxcn/devtools-archlinuxcn/0004-Handle-ArchLinuxARM-server-URL-in-arch-nspawn.patch:
```diff
@@ -0,0 +1,53 @@
++else
++	is_arm=0
++fi
++if ((is_arm)); then
```
I think it is better not to use intermediate variables for such a conditional judgment that has a clear meaning and is only used once?

yuyichao replied:
That's fair. An earlier version had this used elsewhere before I decided that it's easier to just set all the variables here....

yuyichao commented:
> Could you wait me for a few more days before merging this?

Sure. That's why this was submitted as a PR.

Personally, I mainly care about the arm patch and https://github.com/archlinux/devtools/commit/30ed6920c7e7962b7161bc2405c63e9a1ab95be2 .

yan12125 commented:
Here is an issue needing discussions: #2798. It's quite long and not directly related to ARM, so I created a new issue.

CoelacanthusHex commented:
unshare pacstrap patch has been merged upstream.
We can remove it in the near future.
https://gitlab.archlinux.org/archlinux/devtools/-/merge_requests/90

yan12125 commented:
Upstream devtools 20220609 is out :tada:

yuyichao commented:
Updated to be based on 20220609 and I've added patch to implement a overwritable BUILDTOOL and also tweaked to make sure we can track upstream BUILDTOOLVER .

yan12125 commented:
Thanks for the update, nice `$BUILDTOOLVER` trick! I have some other minor changes. Do you mind if I push to `devtools-fixarm` branch directly?

yuyichao commented:
Please do.

yan12125 commented:
Done! I keep additional changes as a separate commit (f0e9666e1e30ab983e643d305ed7085b08a4b8d8). It's better to squash it before merge.

yan12125 commented:
Hmm, just noticed upstream is also discussing how to allow overriding `$BUILDTOOL`: https://gitlab.archlinux.org/archlinux/devtools/-/issues/76. The proposed patch there does not fully match our needs, though.

yan12125 commented:
I think this can be merged?

Just to clarify: I'm not going to rebase 0004-Change-default-BUILDTOOL-to-devtools-archlinuxcn.patch on top of https://gitlab.archlinux.org/archlinux/devtools/-/issues/76 as the latter does not fit our needs.

yuyichao commented:
The current version still looks good to me. I guess if no one else has any comment I can merge sometime soon.

yuyichao commented:
> I guess sudoers on build.archlinuxcn.org should be updated if this pull request is merged as-is.

And I assume that's only necessary to allow someone to change BUILDTOOL.

yuyichao commented:
I was just doing a local test before merging and noticed that the checksums were out of data. Should be fixed now.

yuyichao commented:
It'll also be good if an admin could create a fork of devtools and track the patches there.

yuyichao commented on archlinuxcn/devtools-archlinuxcn/lilac.yaml:
```diff
@@ -3,4 +3,10 @@
+
+update_on:
+  - source: manual
+    manual: 20220207
```
Forgot to update this..... I'll leave it to the next person updating this package to worry about this in a few months....

yan12125 commented:
> And I assume that's only necessary to allow someone to change BUILDTOOL.

Yep

> noticed that the checksums were out of data

Sorry, I forgot to run updpkgsums again after modifying patches.

> It'll also be good if an admin could create a fork of devtools and track the patches there.

Agreed. I don't have the permission to fork https://github.com/archlinux/devtools as https://github.com/archlinuxcn/devtools, either.

lilydjwg commented:
> > It'll also be good if an admin could create a fork of devtools and track the patches there.
> 
> Agreed. I don't have the permission to fork https://github.com/archlinux/devtools as https://github.com/archlinuxcn/devtools, either.

Forked and added permissions to you.

-- 
Best regards,
lilydjwg

yan12125 commented:
Great, thanks! I pushed current patches to https://github.com/archlinuxcn/devtools/commits/master. Next time devtools-archlinuxcn can switch to that fork. Feel free to update it if you prefer a separate branch for patches.

lilydjwg commented:
> And I assume that's only necessary to allow someone to change BUILDTOOL.

It's needed anyway. I've updated the sudoers file on build machines.
